{{cookiecutter.project_slug}}/scripts/verify_pypi_env_variables.py

```diff
@@ -1,36 +1,98 @@
 import os
 import sys
 
-pypi_url = pypi_name = None
+ENV_VAR_PYPI_REPOSITORY_NAME = "PYPI_REPOSITORY_NAME"
+ENV_VAR_PYPI_REPOSITORY_URL_TEMPLATE = "POETRY_REPOSITORIES_{}_URL"
+ENV_VAR_PYPI_REPOSITORY_USERNAME_TEMPLATE = "POETRY_HTTP_BASIC_{}_USERNAME"
+ENV_VAR_PYPI_REPOSITORY_PASSWORD_TEMPLATE = "POETRY_HTTP_BASIC_{}_PASSWORD"
+ENV_VAR_PYPI_REPOSITORY_API_TOKEN_TEMPLATE = "POETRY_PYPI_TOKEN_{}"
 
+pypi_name = "pypi"
+pypi_url = pypi_usename = pypi_password = pypi_api_token = None
+
+exit_code = 0
+
+try:
+    pypi_name = os.environ[ENV_VAR_PYPI_REPOSITORY_NAME]
+except KeyError:
+    pass
+
+print(f"Target pypi repository: '{pypi_name}'")
+
+
+def format_pypi_name(name):
+    return name.replace(".", "_").replace("-", "_").upper()
+
+
+env_var_pypi_repository_url = ENV_VAR_PYPI_REPOSITORY_URL_TEMPLATE.format(
+    format_pypi_name(pypi_name)
+)
+env_var_pypi_repository_username = ENV_VAR_PYPI_REPOSITORY_USERNAME_TEMPLATE.format(
+    format_pypi_name(pypi_name)
+)
+env_var_pypi_repository_password = ENV_VAR_PYPI_REPOSITORY_PASSWORD_TEMPLATE.format(
+    format_pypi_name(pypi_name)
+)
+env_var_pypi_repository_api_token = ENV_VAR_PYPI_REPOSITORY_API_TOKEN_TEMPLATE.format(
+    format_pypi_name(pypi_name)
+)
 try:
-    pypi_url = os.environ["PYPI_REPOSITORY_URL"]
+    pypi_url = os.environ[env_var_pypi_repository_url]
 except KeyError:
     pass
 try:
-    pypi_name = os.environ["PYPI_REPOSITORY_NAME"]
+    pypi_usename = os.environ[env_var_pypi_repository_username]
+except KeyError:
+    pass
+try:
+    pypi_password = os.environ[env_var_pypi_repository_password]
+except KeyError:
+    pass
+try:
+    pypi_api_token = os.environ[env_var_pypi_repository_api_token]
 except KeyError:
     pass
 
-if not pypi_name or pypi_name == "pypi":
+print(pypi_url)
+print(pypi_usename)
+print(pypi_password)
+print(pypi_api_token)
+
+if pypi_name == "pypi":
     if pypi_url:
         print(
-            f"ERROR: (PYPI_REPOSITORY_NAME = '{pypi_name}', "
-            f"PYPI_REPOSITORY_URL = '{pypi_url}') "
-            f"When the environment variable PYPI_REPOSITORY_NAME is 'pypi', "
-            f"empty or not defined, you must NOT define environment variable "
-            f"PYPI_REPOSITORY_URL."
+            f"ERROR: ({ENV_VAR_PYPI_REPOSITORY_NAME} = '{pypi_name}', "
```

Comment:
Je me posais la question des upper cases dans un cas et lower case dans l'autre. Ici ça porte vraiment à confusion parce qu'on a l'impression que ça rend ces variables vraiment différentes.
=> peu importe le choix, fais le même pour toutes tes variables d'environnement : soit tout uppercase soit tout lower case

Reply:
all in lower case